src/CMakeLists.txt

```diff
@@ -623,12 +621,20 @@ if(${WITH_RADOSGW})
     expat
     fcgi
   )
-endif(${WITH_RADOSGW})
-if(${WITH_RBD})
+endif(WITH_RADOSGW)
+if(WITH_RBD)
   set(DENCODER_EXTRALIBS
       ${DENCODER_EXTRALIBS}
       rbd_replay_types)
-endif(${WITH_RBD})
+  list(APPEND dencoder_srcs
+      $<TARGET_OBJECTS:krbd_objs>)
+endif(WITH_RBD)
+if(WITH_KVS)
```

Comment:
@tchaikov 
For now I chose to put it under WITH_RBD, but krbd is even more tricky since that is never going to run on FreeBSD. It is all Linux code.
So perhaps even wrap it in if(LINUX)??
But I was going to delay that until I started working on RBD in general.